modules/terraform-aws-vpc/defaults.tf

```diff
@@ -85,4 +85,26 @@ locals {
     : false),
   false)
 
+  # Determine the output Ids based on create_vpc flag and deployment_template
+  vpc_id = (var.create_vpc ?
+  module.vpc[0].vpc_id : var.existing_vpc_id)
+
+  # If module creates the vpc:
+  # * and it's a cdp vpc with private deployment template public subnets are always empty
+  # * otherwise (i.e. not private deployment or cdp vpc ids from created vpc
+  # If module doesn't create vpc:
+  # * using existing ids
+  public_subnet_ids = (var.create_vpc ?
+    (var.cdp_vpc && (var.deployment_template == "private") ? [] : module.vpc[0].public_subnets)
+  : var.existing_public_subnet_ids)
+
+  private_subnet_ids = (var.create_vpc ?
+    module.vpc[0].private_subnets : var.existing_private_subnet_ids
+  )
+
+  # Currently we only know the RT info if we create the VPC. Further work needed to lookup these resources when create_vpc is false.
```

Comment:
@jimright Why is the route table information required? Is this for validation purposes?
If create\_vpc is true, what route table rules are we creating?

Reply:
>Why is the route table information required? Is this for validation purposes?

> If create_vpc is true, what route table rules are we creating?

The routes created depends on the deployment_template and hence the values of the public_subnets and enable_nat_gateway that are passed to the underlying VPC creation module. See the [underlying terraform-aws-vpc module](https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/v3.19.0/main.tf) for "aws_route"

> Why is the route table information required? Is this for validation purposes?

The only time we use the route table in the aws pre-reqs module is for the Gateway VPC endpoint (see https://github.com/cloudera-labs/terraform-cdp-modules/blob/42308bd0dd3d51e1100caf6a2457bb261b45703b/modules/terraform-cdp-aws-pre-reqs/main.tf#L188-L201). However this is a very useful output to have and used it to setup routes in our Advanced Networking configurations.
Yes, I also think it will be useful parameter for validations.